core/src/rust/filodb_core/src/reader.rs

```diff
@@ -0,0 +1,560 @@
+//! Methods related to reading / querying the index
+
+use std::sync::atomic::Ordering;
+
+use hashbrown::HashSet;
+use jni::{
+    objects::{JByteArray, JClass, JIntArray, JObject, JString, JValue},
+    sys::{jbyteArray, jint, jintArray, jlong, jlongArray, jobjectArray},
+    JNIEnv,
+};
+use tantivy::{collector::FacetCollector, schema::FieldType};
+use tantivy_utils::collectors::part_key_record_collector::PartKeyRecordCollector;
+use tantivy_utils::collectors::string_field_collector::StringFieldCollector;
+use tantivy_utils::collectors::time_collector::TimeCollector;
+use tantivy_utils::collectors::time_range_filter::TimeRangeFilter;
+use tantivy_utils::collectors::{
+    limited_collector::UnlimitedCollector, part_id_collector::PartIdCollector,
+};
+use tantivy_utils::collectors::{
+    part_key_collector::PartKeyCollector, part_key_record_collector::PartKeyRecord,
+};
+use tantivy_utils::field_constants::{self, facet_field_name};
+
+use crate::{
+    errors::{JavaException, JavaResult},
+    exec::jni_exec,
+    jnienv::JNIEnvExt,
+    query::filodb_query::FiloDBQuery,
+    state::IndexHandle,
+};
+
+const TERM_INFO_CLASS: &str = "filodb/core/memstore/TermInfo";
```

Comment:
Though I don't expect to, but if someone refactors and changes the packaging.  Would there be a  compilation failure or there will be a is a runtime exception when running using Tantivy?

Reply:
It would be a runtime exception for ClassNameNotFound or something similar, caught via unit tests.